lib/runtime/offchain/httpset.go

```diff
@@ -0,0 +1,62 @@
+package offchain
+
+import (
+	"net/http"
+	"sync"
+)
+
+var (
+	once    sync.Once
+	HTTPSet *Set
+
+	_ = OnceHTTPSet()
+)
+
+type Set struct {
+	mtx  *sync.Mutex
+	reqs []*http.Request
+}
+
+// OnceHTTPSet
+func OnceHTTPSet() *Set {
+	once.Do(func() {
+		HTTPSet = &Set{
+			mtx:  new(sync.Mutex),
+			reqs: make([]*http.Request, 0),
+		}
+	})
+
+	return HTTPSet
+}
+
+// StartRequest create a new request using the method and the uri, adds the request into the list
+// and then return the position of the request inside the list
+func (p *Set) StartRequest(method, uri string) (int16, error) {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	req, err := http.NewRequest(method, uri, nil)
+	if err != nil {
+		return 0, err
+	}
+
+	p.reqs = append(p.reqs, req)
+	return int16(len(p.reqs) - 1), nil
+}
+
+// Remove just remove a expecific request from reqs
+func (p *Set) Remove(id int) {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	p.reqs = append(p.reqs[:id], p.reqs[id+1:]...)
```

Comment:
We should use a map instead of a slice because when we are removing we are shifting the index and if we call get again after removal we will get an incorrect index.
```
arr : [1,2,3,4,5,6]
Remove(2)
arr: [1,2,4,5,6]
Get(3) will return 5 instead of 4.
```

Reply:
Also, the `Get` logic of length checking will fail.